File: d66-social/src/main/java/com/eroldmr/d66/appuser/verificationtoken/VerificationToken.java
package com.eroldmr.d66.appuser.verificationtoken;

import com.eroldmr.d66.appuser.AppUser;
import lombok.*;

import javax.persistence.*;

import java.time.Instant;
import java.time.LocalDateTime;

import static javax.persistence.FetchType.LAZY;

/**
 * @author Mompati 'Patco' Keetile
 * @created 19-12-2022 @ 12:04
 */
@Getter @Setter @AllArgsConstructor @NoArgsConstructor @Builder(builderMethodName = "NewToken")
@Entity
@Table(name = "token")
public class VerificationToken {
  @Id
  @SequenceGenerator(name = "token_id_seq", sequenceName = "token_id_seq", allocationSize = 1)
  @GeneratedValue(generator = "token_id_seq", strategy = GenerationType.IDENTITY)
  private Long id;
  private String token;
  @ManyToOne(fetch = LAZY)
  private AppUser appUser;
  private LocalDateTime expiresAt;
}

File: d66-social/src/main/kotlin/com/eroldme/d66/user/register/token/VerificationToken.kt
package com.eroldme.d66.user.register.token

import com.eroldme.d66.user.ApplicationUser
import lombok.AllArgsConstructor
import lombok.Builder
import lombok.Data
import lombok.NoArgsConstructor
import java.time.Instant
import javax.persistence.*
import javax.persistence.FetchType.LAZY

/**
 * @author Mompati 'Patco' Keetile
 * @created 06-12-2022 @ 14:11
 *
 *
 */
@Data
@Entity
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "token")
class VerificationToken(
    @Id
    @SequenceGenerator(name = "token_id_seq", sequenceName = "token_id_seq", allocationSize = 1)
    @GeneratedValue(generator = "token_id_seq", strategy = GenerationType.IDENTITY)
    private val id: Long,
    private val token: String,
    @ManyToOne(fetch = LAZY)
    private val applicationUser: ApplicationUser,
    private val expiryDate: Instant
)